### PR TITLE
[Duplicate of #7519] Page Transitions in React using Framer Motion

### LINKED ISSUE #7321: Page Transitions in React using Framer Motion
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

**NOTE: Any topic suggestion copied/taken/modified from YouTube may Lead to a permanent ban from the community.**

### Topic Suggestion
Enter topic suggestion here...

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

### Proposal Submission

####[Languages] Page Transitions in React using Framer Motion
 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
*Framer Motion is an animation library that makes creating animations easy. Its simplified API helps us abstract the complexities behind animations and allows us to create animations with ease. Un this article we'll  be using framer motion to make page transitions in React*
.

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
At the end of this tutorial, the readers would
- Have an understanding about framer motion and how it works
- know how how to use `animatepresence` in framer motion as that is what we will be using in this tutorial
- make page transitions using framer motion.

#### Article quality
other articles that I've come across just go straight to building not minding if the user has knowledge about the article or not. In my article, I'll be discussing the terms needed for the users to follow through and also stating the prerequisites for the tutorial. I'll break down the some terms in the documentation to make it easier for the readers to understand.

#### References
(https://www.framer.com/docs/)

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by oyedeletemitope**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #7321 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
#7519

**Comment by hectorkambow** [comment]
image

**Comment by oyedeletemitope** [comment]
Hello @hectorkambow is there anything I did wrong while submitting?

**Comment by hectorkambow** [comment]
Possibly the repo was not forked properly - be sure to refer to our docs for more details. 👍

**Comment by oyedeletemitope** [comment]
Hello @hectorkambow  I tried making some changes if the error still persists then I think I’ll have clone the repo again and close this pr

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
+The React.js framework allows you to create single-page applications. These are applications in which the pages won’t reload when you click on different links. The transition from one link to the next is so quick that it's difficult to notice when the content changes. As a result, we can use transitions while navigating through various links or routes on a website. This improves the application's User Experience.
+
+For some, transitioning from page to page has often seemed difficult. We can archive that using a library called framer motion.
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:3]: What is the objective of this tutorial? State what is going to be covered here

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+## What is framer motion?
+
+Framer Motion is an animation library that makes it so simple to create animations. Its simplified API allows us to extract the complexities of animations. Thus creating animations with ease. An open-source, production-ready React motion library. Framer Motion is ideal for dealing with animations on 99 percent of websites.
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:6]: Add link to Framer motion

Comment (srishilesh) [reply content/articles/page-transitions-in-react-using-framer-motion/index.md:6]: Add any example website that uses framer motion

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
+The React.js framework allows you to create single-page applications. These are applications in which the pages won’t reload when you click on different links. The transition from one link to the next is so quick that it's difficult to notice when the content changes. As a result, we can use transitions while navigating through various links or routes on a website. This improves the application's User Experience.
+
+For some, transitioning from page to page has often seemed difficult. We can archive that using a library called framer motion.
+## What is framer motion?
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:4]: Start the markdown header from `###`

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+Framer Motion is an animation library that makes it so simple to create animations. Its simplified API allows us to extract the complexities of animations. Thus creating animations with ease. An open-source, production-ready React motion library. Framer Motion is ideal for dealing with animations on 99 percent of websites.
+
+The `motion` component is the Framer motion core API. Every HTML and SVG element has a `motion` component. They function exactly the same as their HTML counterparts. They also have more props that allow the addition of animations and gestures. . We would get to understand better when we’re building our project. So let’s get started
+## Prerequisites
+
+To follow through this article, we need to have a basic understanding of Javascript. We also need to know how react-router works.
+## Page Transitions With Framer-motion
+
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:13]: What are we going to build? State the objective of the code project.

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+          </div>
+        </div>
+      </nav>
+    </>
+  );
+};
+
+export default Navbar;
+```
+
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:174]: Add explanation of the above code as bulleted points.

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+        <Route path="/contact" element={<Contact />} />
+      </Routes>
+    </AnimatePresence>
+  );
+}
+
+export default AnimatedRoutes;
+```
+
+Here, we imported our `Routes`, `Route`, and `useLocation` from react-router-dom. `Routes` are a significant improvement over the `Switch` element in v5. The route will allow us to go to each of the pages we created in response to our request. The 'useLocation' hook will assist our routes in determining which pages we are on. We also imported our pages as well as the `animatedPresence` from framer motion`. We then wrapped our routes with our `animatePresence`. We’re almost done with working on our pages folder.
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:203]: Make this as bulleted points.

File: content/articles/page-transitions-in-react-using-framer-motion/index.md [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+**NOTE: we didn’t work on our index.js, we only edited it to look like this:**
+![work](/engineering-education/page-transition-in-react-using-framer-motion/work.png)
+
+Let’s now check out our result in the browser:
+![result](/engineering-education/page-transition-in-react-using-framer-motion/result.gif)
+
+Here’s a [link](https://github.com/oyedeletemitope/page-transition-in-react-using-framer-motion) to the github repository.
+
+## Conclusion
+Our main discussion was about framer motion as a tool for making page transitions in React. We also were able to make page transitions by making use of it in our project. Would you please share if this was helpful?
```
Comment (srishilesh) [inline content/articles/page-transitions-in-react-using-framer-motion/index.md:334]: Add any reference links that might help the reader

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @oyedeletemitope ,

I've suggested some minor changes. Please go through them carefully, and ping me when you're done.

Thanks.

**Comment by oyedeletemitope** [comment]
@srishilesh changes made as requested

**Review by srishilesh — APPROVED** [review]
Hey @oyedeletemitope ,

Make sure to go through the changes I've made.

Moving for the final review.

Thanks.

**Comment by oyedeletemitope** [comment]
Thanks @srishilesh i’ll go through them

**Comment by hectorkambow** [comment]
@oyedeletemitope 

well done - I'd recommend you add a skill section to your author profile. 

This will go a long way for future potential employers.

**Comment by oyedeletemitope** [comment]
Thank you @hectorkambow i’ll be sure to do that next time

**Comment by hectorkambow** [comment]
👍  

Fantastic - once the article is live (to help it gain more traction) be sure to share it on LinkedIn & Twitter (and tag us if you could). 

This will give your freshly published article a nice boost of engagement. 

Also note you can see the # of page view your articles get via the EngEd site.

**Pull request merged**